### PR TITLE
refactor workflow logs

### PR DESCRIPTION
**Pull request by Yancey0623**
The PR refactor the workflow log content as an example:

``` text
Step [1/3] Execute Code: echo hello1
Step [1/3] Log view: http://localhost:8001/workflows/default/steps-bdpff?nodeId=steps-bdpff-xx1
Step [1/3] Status: Pending
Step [1/3] Status: Running
Step [1/3] Status: Succeed/Failed
Step [2/3] Execute Code: echo hello2
Step [2/3] Log view: http://localhost:8001/workflows/default/steps-bdpff?nodeId=steps-bdpff-xx2
...
```

### CODE REVIEW

File: pkg/argo/fetch.go [hunk 1]
```diff
@@ -40,6 +43,32 @@ func newFetchResponse(newReq *pb.FetchRequest, eof bool, logs []string) *pb.Fetc
...
+		if e != nil {
+			return -1, e
+		}
+	}
+}
+
+func logViewURL(ns, wfID, stepID string) (string, error) {
+	ep := os.Getenv("SQLFLOW_ARGO_UI_ENDPOINT")
+	if ep == "" {
+		return "", fmt.Errorf("should set SQLFLOW_ARGO_UI_ENDPOINT if enable Argo mode")
```
Comment (typhoonzero) [inline pkg/argo/fetch.go]: Maybe raise this error in some early stage before the job have started running.

File: pkg/argo/fetch.go [hunk 2]
```diff
@@ -55,48 +84,63 @@ func Fetch(req *pb.FetchRequest) (*pb.FetchResponse, error) {
...
-
-	// finish fetching the current pod logs when:
-	// 1. the Pod is completed(succeed/failed/error).
-	// 2. no updated logs.
-	if req.LogOffset == newLogOffset && isPodCompleted(pod) {
-		finishFetchingCurrentPod = true
+	// output the refreshed status if the Pod Phase changed.
+	if req.LogOffset != string(pod.Status.Phase) {
+		logs = append(logs, fmt.Sprintf("Step: [%d/%d] Status: %s", stepIdx, stepCnt, pod.Status.Phase))
+		newOffset = string(pod.Status.Phase)
```
Comment (typhoonzero) [inline pkg/argo/fetch.go]: Why the offset is set to `phase`?

Comment (Yancey0623) [reply pkg/argo/fetch.go]: > Use the Pod phase as the offset to avoid output the large duplicated logs.

I add some comment in the code.

Comment (Yancey0623) [reply pkg/argo/fetch.go]: Renamed  `logOffset` to `stepPhase` which better to understand.

**Review by typhoonzero — APPROVED** [review]
LGTM, do we need to update the client also?

**Comment by Yancey0623** [comment]
> LGTM, do we need to update the client also?

@typhoonzero Yes, I will update pysqlflow recently.

**Comment by weiguoz** [comment]
I will update the [jsqlflow](https://github.com/sql-machine-learning/jsqlflow) if this PR is merged.

**Pull request merged**